### PR TITLE
Cross-table feature

### PR DESCRIPTION
**Pull request by beaugogh**

### CODE REVIEW

File: src/app/modules/gb-data-selection-module/gb-data-selection.component.html [hunk 1]
```diff
@@ -99,29 +99,29 @@
...
+              <!--<span *ngIf="queryService.showObservationCounts"-->
+                    <!--class="gb-data-selection-emphasis-text">-->
+                <!--{{observationCount_2}} observations-->
+              <!--</span>-->
+            <!--</div>-->
+          <!--</div>-->
+        <!--</div>-->
+      <!--</md2-accordion-header>-->
+      <!--<gb-export></gb-export>-->
+    <!--</md2-accordion-tab>-->
```
Comment (ewelinagr) [inline src/app/modules/gb-data-selection-module/gb-data-selection.component.html]: Why is the export step commented out?

File: src/app/config/env.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-  "env": "default"
+  "env": "dev"
```
Comment (ewelinagr) [inline src/app/config/env.json]: this should stay "default", right?

File: src/app/modules/gb-data-selection-module/gb-data-selection.component.html (hunk 1 above)
Comment (ewelinagr) [reply src/app/modules/gb-data-selection-module/gb-data-selection.component.html]: Ok, I see that export will be moved somewhere else and this part here will be removed, right?

File: src/app/services/cross-table.service.ts [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+        let hasIt = this.headerConstraints.has(con);
+        if (!hasIt) {
+          mapped = hasIt;
+        }
+      });
+    }
+    return mapped;
+  }
+
+  mockDataInit() {
```
Comment (ewelinagr) [inline src/app/services/cross-table.service.ts]: this part is now duplicated in cross-table.service.mock.ts

Comment (beaugogh) [reply src/app/services/cross-table.service.ts]: yes, the two are rather similar, but one is using dimensions the other is using constraints

File: src/app/config/env.json (hunk 2 above)
Comment (beaugogh) [reply src/app/config/env.json]: yes, will change this

File: src/app/modules/gb-data-selection-module/gb-data-selection.component.html (hunk 1 above)
Comment (beaugogh) [reply src/app/modules/gb-data-selection-module/gb-data-selection.component.html]: will remove this

File: yarn-error.log [hunk 4]
```diff
@@ -6605,14 +6605,11 @@ Lockfile:
...
       at step (/Users/bo/.nvm/versions/node/v8.10.0/lib/node_modules/yarn/lib/cli.js:98:30)
-      at /Users/bo/.nvm/versions/node/v8.10.0/lib/node_modules/yarn/lib/cli.js:116:14
-      at new Promise (<anonymous>)
-      at new F (/Users/bo/.nvm/versions/node/v8.10.0/lib/node_modules/yarn/lib/cli.js:23451:28)
-      at /Users/bo/.nvm/versions/node/v8.10.0/lib/node_modules/yarn/lib/cli.js:95:12
-      at /Users/bo/.nvm/versions/node/v8.10.0/lib/node_modules/yarn/lib/cli.js:36660:25
-      at Array.map (<anonymous>)
+      at /Users/bo/.nvm/versions/node/v8.10.0/lib/node_modules/yarn/lib/cli.js:109:13
+      at <anonymous>
+      at process._tickCallback (internal/process/next_tick.js:188:7)
```
Comment (ewelinagr) [inline yarn-error.log]: maybe we should include the whole 'yarn-error.log' in the .gitignore file?

Comment (beaugogh) [reply yarn-error.log]: good idea, will do!

File: src/app/modules/gb-data-selection-module/gb-data-selection.component.html [hunk 5]
```diff
@@ -39,8 +39,8 @@
       <gb-selection></gb-selection>
     </md2-accordion-tab>
 
-    <!-- The 2nd step -->
-    <md2-accordion-tab [active]="false">
+    <!-- The 2nd step: variable selection -->
+    <md2-accordion-tab [active]="true">
```
Comment (ewelinagr) [inline src/app/modules/gb-data-selection-module/gb-data-selection.component.html]: is it a change on purpose (second step should be active by default) or was added just for testing?

Comment (beaugogh) [reply src/app/modules/gb-data-selection-module/gb-data-selection.component.html]: sorry, just added for testing, will change it back

**Comment by beaugogh** [comment]
@ewelinagr I think it's ready to merge

**Pull request merged**